src/pages/home/report/ReportActionCompose.js

```diff
@@ -165,1165 +178,1167 @@ const getMaxArrowIndex = (numRows, isAutoSuggestionPickerLarge) => {
+    const [isFullComposerAvailable, setIsFullComposerAvailable] = useState(isComposerFullSize);
+
+    const isEmptyChat = useMemo(() => _.size(reportActions) === 1, [reportActions]);
 
-    setIsFullComposerAvailable(isFullComposerAvailable) {
-        this.setState({isFullComposerAvailable});
-    }
+    const shouldAutoFocus = !modal.isVisible && (shouldFocusInputOnScreenFocus || isEmptyChat) && shouldShowComposeInput;
+
+    // These variables are used to decide whether to block the suggestions list from showing to prevent flickering
+    const shouldBlockEmojiCalc = useRef(false);
+    const shouldBlockMentionCalc = useRef(false);
 
     /**
      * Updates the should clear state of the composer
-     *
-     * @param {Boolean} shouldClear
      */
-    setTextInputShouldClear(shouldClear) {
-        this.setState({textInputShouldClear: shouldClear});
-    }
+    const [textInputShouldClear, setTextInputShouldClear] = useState(false);
+    const [isCommentEmpty, setIsCommentEmpty] = useState(comment.length === 0);
 
     /**
      * Updates the visibility state of the menu
-     *
-     * @param {Boolean} isMenuVisible
      */
-    setMenuVisibility(isMenuVisible) {
-        this.setState({isMenuVisible});
-    }
+    const [isMenuVisible, setMenuVisibility] = useState(false);
+    const [selection, setSelection] = useState({
+        start: isMobileSafari && !shouldAutoFocus ? 0 : comment.length,
+        end: isMobileSafari && !shouldAutoFocus ? 0 : comment.length,
+    });
+    const [value, setValue] = useState(comment);
+
+    const [composerHeight, setComposerHeight] = useState(0);
+    const [isAttachmentPreviewActive, setIsAttachmentPreviewActive] = useState(false);
+
+    // TODO: rewrite suggestion logic to some hook or state machine or util or something to not make it depend on ReportActionComposer
+    const [suggestionValues, setSuggestionValues] = useState(defaultSuggestionsValues);
+
+    const isEmojiSuggestionsMenuVisible = !_.isEmpty(suggestionValues.suggestedEmojis) && suggestionValues.shouldShowEmojiSuggestionMenu;
+    const isMentionSuggestionsMenuVisible = !_.isEmpty(suggestionValues.suggestedMentions) && suggestionValues.shouldShowMentionSuggestionMenu;
+
+    const [highlightedEmojiIndex] = useArrowKeyFocusManager({
+        isActive: isEmojiSuggestionsMenuVisible,
+        maxIndex: getMaxArrowIndex(suggestionValues.suggestedEmojis.length, suggestionValues.isAutoSuggestionPickerLarge),
+        shouldExcludeTextAreaNodes: false,
+    });
+    const [highlightedMentionIndex] = useArrowKeyFocusManager({
+        isActive: isMentionSuggestionsMenuVisible,
+        maxIndex: getMaxArrowIndex(suggestionValues.suggestedMentions.length, suggestionValues.isAutoSuggestionPickerLarge),
+        shouldExcludeTextAreaNodes: false,
+    });
+
+    const insertedEmojis = useRef([]);
```

Comment:
I would rename this to `insertedEmojisRef` to make obvious with what data structure we are dealing here (because we misused it later on, check further comments)

Reply:
I fixed that :v: